src/mgba_printf/mini_printf.c

```diff
@@ -275,10 +382,24 @@ s32 mini_vpprintf(s32 (*puts)(char* s, s32 len, void* buf), void* buf, const cha
                         len = puts(ptr, len, buf);
                     }
                     break;
+                case 'S' :
+                    ptr = va_arg(va, char*);
+                    len = mini_strlen2(ptr);
+                    if (pad_to > 0)
+                    {
+                        len = mini_pad(ptr, len, pad_char, pad_to, bf);
+                        len = puts2(bf, len, buf);
+                    } else
+                    {
+                        len = puts2(ptr, len, buf);
+                    }
+                    break;
                 default:
                     len = 1;
                     len = puts(&ch, len, buf);
                     break;
+                    // MODIFIED: Use %S to convert to Ascii
```

Comment:
Comment is not really necessary, if the function shouod be documented, it should be done at the top